deepmd/pt/model/task/fitting.py

```diff
@@ -500,10 +500,30 @@ def _forward_common(
             assert self.aparam_inv_std is not None
             if aparam.shape[-1] != self.numb_aparam:
                 raise ValueError(
-                    "get an input aparam of dim {aparam.shape[-1]}, ",
-                    "which is not consistent with {self.numb_aparam}.",
+                    f"get an input aparam of dim {aparam.shape[-1]}, ",
+                    f"which is not consistent with {self.numb_aparam}.",
+                )
+            aparam = aparam.view([nf, -1, self.numb_aparam])
+            natom = aparam.shape[1]
+            if natom == nloc:  # good
+                pass
+            elif natom < nloc:  # for spin or other methods with virtual atoms
```

Comment:
aparam should be augmented at the model level rather than being managed inside the fitting.